drv/Maps.drv

```diff
@@ -324,11 +317,112 @@ public class MAPS {
 		public int hashCode() { synchronized(sync) { return map.hashCode(); } }
 
 		@Override
-		public boolean equals(final Object o) { if (o == this) return true; synchronized(sync) { return map.equals(o); } }
+		public boolean equals(final Object o) {
+			if (o == this) return true;
+			if (o instanceof SynchronizedMap) synchronized (sync) { return map.equals(((SynchronizedMap) o).map); }
```

Comment:
VERY DANGEROUS. Accesses a synchronized map without synchronization! 

In this cases, a look at the JDK code is a good idea...